src/ddscxx/src/org/eclipse/cyclonedds/topic/qos/TopicQosDelegate.cpp

```diff
@@ -226,7 +236,8 @@ TopicQosDelegate::operator ==(const TopicQosDelegate& other) const
            other.resources_   == resources_   &&
            other.priority_    == priority_    &&
            other.lifespan_    == lifespan_    &&
-           other.ownership_   == ownership_;
+           other.ownership_   == ownership_   &&
+           other.datarepresentation_ == datarepresentation_;
```

Comment:
Please also encapsulate this line with the conditional OMG_DDS_EXTENSIBLE_AND_DYNAMIC_TOPIC_TYPE_SUPPORT macro. (BTW: I see that in this particular case the embedding OMG_DDS_OWNERSHIP_SUPPORT macro was also missing for the "ownership" policy).

Reply:
Will do.